Added new experimental data

hung-chi opened the pull request:
- [x] Tests pass
- [x] Appropriate changes to README are included in PR

hung-chi committed:
Added nfet01v8 pfet01v8 meas

hung-chi committed:
Added nb for sky130 meas

hung-chi committed:
Updated experimental data

hung-chi committed:
Debugged README

Signed-off-by: hung-chi <85979806+hung-chi@users.noreply.github.com>

hung-chi committed:
Update README.rst

Signed-off-by: hung-chi <85979806+hung-chi@users.noreply.github.com>

msaligane commented:
@hung-chi Can you add the notebook you showed me last Friday?

mithro commented:
@hung-chi - Is that notebooks part of this pull request?

hung-chi commented:
> @hung-chi - Is that notebooks part of this pull request?

Yes, the notebook is included. You could find it in my forked repo at https://github.com/hung-chi/skywater-pdk-sky130-raw-data/blob/develop/notebooks/electrical_characterization_sky130_epfl.ipynb

hung-chi committed:
Added Gm/ID plot to notebook

hung-chi commented:
@msaligane 

I added the Gm/ID plot. You can find them in the updated [notebook](https://github.com/hung-chi/skywater-pdk-sky130-raw-data/blob/develop/notebooks/electrical_characterization_sky130_epfl.ipynb).

hung-chi commented:
#23

proppy commented on README.rst:
```diff
@@ -10,6 +10,9 @@ Experimental Data
 
+**Other data**
+
+* (2022-Nov-1) New experimental data by EPFL is stored in the `.csv` format at `sky130_fd_pr/epfl <./sky130_fd_pr/epfl>`_. The corresponded notebook can be found at `notebooks/electrical_characterization_sky130_epfl.ipynb <./notebooks/electrical_characterization_sky130_epfl.ipynb>`_.
```
maybe move the data under a different top level dirrectory for consistency?
```
sky130_fd_pr_epfl/cells
```
@mithro any input there?

hung-chi replied:
Got it

hung-chi replied:
Fixed in [be9ac42](https://github.com/google/skywater-pdk-sky130-raw-data/pull/23/commits/be9ac42dd47947dc63d815e460dea4e5a2586848)

proppy commented on notebooks/electrical_characterization_sky130_epfl.ipynb:
```diff
@@ -0,0 +1,1204 @@
+{
+ "cells": [
```
can we add a new cell with a colab badge:
```
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "view-in-github",
+    "colab_type": "text"
+   },
+   "source": [
+    "<a href=\"https://colab.research.google.com/github/proppy/skywater-pdk-sky130-raw-data/blob/develop/notebooks/electrical_characterization_sky130_epfl.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
+   ]
+  },
```
and another one that clone the repo, similar to what we have for the other notebooks:
https://github.com/google/skywater-pdk-sky130-raw-data/blob/main/notebooks/sky130_plot_gm.ipynb
https://github.com/google/skywater-pdk-sky130-raw-data/blob/main/notebooks/sky130_plot_current.ipynb

also maybe rename it to have s similar naming scheme as the other one?

@bmurmann should we look into consolidating with your existing notebook and allow it to work across multiple dataset?

bmurmann replied:
My notebook is very simple, best to build on this new one instead of consolidating/merging.

hung-chi replied:
Got it. Notebooks were renamed in [04d30b1](https://github.com/google/skywater-pdk-sky130-raw-data/pull/23/commits/04d30b1393ecb968d54b4ff09c54503879b9db61).
Also, README was updated correspondingly in [f6bc6ab](https://github.com/google/skywater-pdk-sky130-raw-data/pull/23/commits/f6bc6ab632dfe5e4db7e2f77cf0c839ce1d0708d)

proppy reviewed (approved):
sorry for the late feedback!

Just a few minor cosmetic changes.

hung-chi committed:
Added read_data.py

hung-chi committed:
Added plot_iv.py

hung-chi committed:
Added nb for plotting data

hung-chi committed:
Added data 5290

hung-chi committed:
Added data 2605

hung-chi committed:
Added data 2602

hung-chi committed:
Updated 2602-1

hung-chi committed:
Added notebook for plotting EPFL data

hung-chi committed:
Improved plotting and reader

hung-chi committed:
Improved nb

hung-chi committed:
Updated nb and plot

hung-chi committed:
Debugged the data and plot

hung-chi committed:
Refactored data location to high level directory

hung-chi committed:
Renamed notebooks

hung-chi committed:
Updated README.rst

proppy commented:
@hung-chi thanks for working on this!

Do you think it would it makes sense to modify the notebook so that they can be opened in https://colab.research.google.com/ ?

If you agree, maybe we can do the follow-up changes (either in this PR or separatly):
- inline the utils in `sky130_plot_epfl_data_general` notebook (they only seem to be used here)
- add a colab badge similar to the other notebooks (see https://github.com/google/skywater-pdk-sky130-raw-data/blob/main/notebooks/sky130_plot_current.ipynb)
- self-clone the repo in the first cell (so that the data is accessible from the hosted notebook environment)

hung-chi committed:
Debugged README.rst

hung-chi commented:
> @hung-chi thanks for working on this!
> 
> Do you think it would it makes sense to modify the notebook so that they can be opened in https://colab.research.google.com/ ?
> 
> If you agree, maybe we can do the follow-up changes (either in this PR or separatly):
> 
> * inline the utils in `sky130_plot_epfl_data_general` notebook (they only seem to be used here)
> * add a colab badge similar to the other notebooks (see https://github.com/google/skywater-pdk-sky130-raw-data/blob/main/notebooks/sky130_plot_current.ipynb)
> * self-clone the repo in the first cell (so that the data is accessible from the hosted notebook environment)

Sure, I can do that in the PR. Also fine with inlining the utils in `sky130_plot_epfl_data_general`  notebook.

hung-chi committed:
Updated nb

1. included in-line utils
2. included colab
3. included copyright and license
4. included git clone

hung-chi commented:
Hi @proppy 

I have updated the notebook by including the self-clone, in-line utils, colab.
The notebook will work correctly after merging.

Thanks

proppy commented on notebooks/sky130_plot_epfl_data_first_meas.ipynb:
```diff
@@ -0,0 +1,1426 @@
+    }
+   },
+   "source": [
+    "<h1>Electrical characterization of SkyWater 130 nm<span class=\"tocSkip\"></span></h1>\n",
```
can you add the colab badge here too?

proppy commented on notebooks/sky130_plot_epfl_data_first_meas.ipynb:
```diff
@@ -0,0 +1,1426 @@
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import sekve\n",
```
I think you may need a few `%pip install` command before you can import this.

proppy commented on notebooks/sky130_plot_epfl_data_general.ipynb:
```diff
@@ -0,0 +1,508 @@
+    "import re\n",
+    "import os\n",
+    "import os.path as op\n",
+    "import sekve\n",
```
I think you may need a few `%pip install` command before you can import this.
